packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -14,10 +14,15 @@ import type { CustomSelectButtonSize } from './types';
 
 const ITEM_PADDING = space( 2 );
 
+export const SelectWrapper = styled.div`
+	display: block;
+`;
+
 export const WithHintWrapper = styled.div`
 	display: flex;
 	justify-content: space-between;
 	flex: 1;
+	flex-wrap: wrap;
```

Comment:
| Before | After |
|---------|---------|
|  <img width="324" alt="Screenshot 2024-06-07 at 12 50 57 p m" src="https://github.com/WordPress/gutenberg/assets/81248/7f51dd60-cbc4-41ac-bb68-fcff23115f26"> |  <img width="334" alt="Screenshot 2024-06-07 at 1 15 14 p m" src="https://github.com/WordPress/gutenberg/assets/81248/68e9c9d2-e234-4c18-a00c-fbd58775585a">
 |

Reply:
While this can be also replicated in the V1 component, I agree that it's something that could be improved and I think that your suggestion makes sense. We can tackle it in a separate issue and update V1 at the same time to keep consistency, in case we think we should address it.

Reply:
Opened https://github.com/WordPress/gutenberg/pull/62848 to tackle this separately